### PR TITLE
Defining defined

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: src/main/resources/standardTestCases/defined_function/defined_function.wdl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      File real_f = select_first([mk_file.f])
+      call cat { input: f = real_f }
+    }
+  }
+  output {
+    # Array[String] dit_out = defined_in_task.out
+    Array[String?] cat_out = cat.out
+    Array[Int] good_indices = select_all(good_index)
+  }
+}
```
Comment (mcovarr) [inline src/main/resources/standardTestCases/defined_function/defined_function.wdl]: \n

File: src/main/resources/standardTestCases/defined_function/defined_function.wdl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  Array[Boolean] mask = [true, false, true, false]
+  Array[Int] indices = range(4)
+  Array[Pair[Boolean, Int]] masked_indices = zip(mask, indices)
+
+  scatter (p in masked_indices) {
+    if (p.left) {
+      Int good_index = p.right
+      call mk_file { input: index = good_index }
+    }
+    # call defined_in_task { input: f = mk_file.f }
```
Comment (mcovarr) [inline src/main/resources/standardTestCases/defined_function/defined_function.wdl]: various commented out stuff should be removed

File: src/main/resources/standardTestCases/defined_function/defined_function.wdl [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    if (defined(mk_file.f)) {
+      File real_f = select_first([mk_file.f])
+      call cat { input: f = real_f }
+    }
+  }
```
Comment (mcovarr) [inline src/main/resources/standardTestCases/defined_function/defined_function.wdl:31]: I'm 😕  what's being tested here

**Review by mcovarr — APPROVED** [review]

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/156/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

**Comment by kshakir** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/156/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/resources/standardTestCases/defined_function/defined_function.wdl (hunk 3 above)
Comment (cjllanwarne) [reply src/main/resources/standardTestCases/defined_function/defined_function.wdl:31]: I want to test `defined` both in the workflow for flow control and in the task. Unfortunately, https://github.com/broadinstitute/cromwell/issues/1982 stops the in-task one from working

File: src/main/resources/standardTestCases/defined_function/defined_function.wdl (hunk 2 above)
Comment (cjllanwarne) [reply src/main/resources/standardTestCases/defined_function/defined_function.wdl]: Hmm, I half agree. I really like that it's easy to re-enable to verify a fix for https://github.com/broadinstitute/cromwell/issues/1982... but yeah I don't like the commenting

Comment (cjllanwarne) [reply src/main/resources/standardTestCases/defined_function/defined_function.wdl]: Also, these things *should* be part of this test, it's just unfortunate that it doesn't work yet

**Pull request merged**